### PR TITLE
Add top-level test directory for "main" tests

### PR DESCRIPTION
**Pull request by mcgibbon**
## Purpose

Our environment and test definitions have been duplicated in several places, due to having multiple packages each with their own copy. This PR consolidates "main" tests on CircleCI into a single job using one test directory. Tests for each package are split into subdirectories of `tests/main` at the top level.

Pace-util tests are kept separate for two reasons - one, to avoid potential issues for the ML team, we want to ensure pace-util works without our other packages installed in the environment. Two, those tests take a fair amount of time to run, and text execution time will be improved by running those tests in parallel.

## Requirements changes:

- added top-level `requirements_dev.txt` used for CircleCI tests

## Infrastructure changes:

- Deleted CircleCI plans test_dsl, test_physics, and test_fv3core, moved main driver tests out of test_driver and into test_main
- added top-level `requirements_dev.txt` used by main CircleCI test. This will be consolidated with our other environment definitions in a future PR.

## Checklist
Before submitting this PR, please make sure:

- [ ] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [ ] Docstrings and type hints are added to new and updated routines, as appropriate
- [ ] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [ ] For each public change and fix in `pace-util`, HISTORY has been updated
- [ ] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

File: .jenkins/install_virtualenv.sh [hunk 1]
```diff
@@ -28,13 +28,6 @@ git submodule update --init ${PACE_DIR}/external/daint_venv
...
 source ${virtualenv_path}/bin/activate
-python3 -m pip install ${PACE_INSTALL_FLAGS} ${PACE_DIR}/external/gt4py/
-python3 -m pip install ${PACE_DIR}/pace-util/
-python3 -m pip install $wheel_command -c ${PACE_DIR}/constraints.txt -r fv3core/requirements/requirements_base.txt
-python3 -m pip install ${PACE_INSTALL_FLAGS} ${PACE_DIR}/fv3core/
-python3 -m pip install ${PACE_INSTALL_FLAGS} ${PACE_DIR}/fv3gfs-physics/
-python3 -m pip install ${PACE_INSTALL_FLAGS} ${PACE_DIR}/stencils/
-python3 -m pip install ${PACE_INSTALL_FLAGS} ${PACE_DIR}/dsl/
-python3 -m pip install ${PACE_INSTALL_FLAGS} ${PACE_DIR}/driver/
+python3 -m pip install $wheel_args -r ${PACE_DIR}/requirements.txt -c ${PACE_DIR}/constraints.txt
```
Comment (mcgibbon) [inline .jenkins/install_virtualenv.sh:31]:
```suggestion
python3 -m pip install $wheel_args -r ${PACE_DIR}/requirements_dev.txt -c ${PACE_DIR}/constraints.txt
```

**Review by jdahm — APPROVED** [review]
This simplifies it nicely. Good work!

**Comment by mcgibbon** [comment]
launch jenkins

**Comment by mcgibbon** [comment]
launch jenkins

**Pull request merged**